packages/grid/src/vaadin-grid-selection-column-mixin.js

```diff
@@ -81,12 +81,16 @@ export const GridSelectionColumnMixin = (superClass) =>
     }
 
     /**
-     * Return true if array `a` contains all the items in `b`
+     * Return true if array `left` contains all the items in `right`,
+     * using the `getItemId` method to compare items.
      * We need this when sorting or to preserve selection after filtering.
      * @private
      */
-    __arrayContains(a, b) {
-      return Array.isArray(a) && Array.isArray(b) && b.every((i) => a.includes(i));
+    __arrayContains(left, right) {
+      left ||= [];
+      right ||= [];
```

Comment:
Could we use default parameters instead?

```suggestion
    __arrayContains(left = [], right = []) {
```

Reply:
Moved the method to selection mixin, at least in the current usage context the fallbacks actually don't seem necessary.